### ISSUE TITLE
Using SSH tunnel

### ISSUE THREAD

**Issue by anteldan** [dbeaver/dbeaver#7553]
#### System information: 
- RHEL 8.1
- DBeaver 6.3.2
- Java:
openjdk version "1.8.0_232"
OpenJDK Runtime Environment (build 1.8.0_232-b09)
OpenJDK 64-Bit Server VM (build 25.232-b09, mixed mode)

#### Connection specification:
- MariaDB 10.3
- org.mariadb.jdbc:mariadb-java-client:RELEASE version 2.4.3
DBeaverData/drivers/maven/maven-central/org.mariadb.jdbc/mariadb-java-client-2.4.3.jar
- SSH tunnel

#### Describe the problem you're observing:
Since DBeaver 6.3.2, I cannot connect to MariaDB server via SSH tunnel and I get SSH banned for a while from that machine, though I only clicked once on "Test tunnel configuration" -> "SSH_MSG_DISCONNECT: 2 Too many authentication failures "
This happened on previous DBeaver version only once and only the first time when creating a new connection, I got banned and then the next attempt was OK.
With DBeaver 6.3.2, I tried 3 times and got banned each time so I quit.

#### Include any warning/errors/backtraces from the logs

Message after trying to open connection :
"Can't initialize tunnel
  Cannot establish tunnnel
  Cannot establish tunnnel
    java.net.ConnectException: Connection refused
    java.net.ConnectException: Connection refused
      Connection refused
"

**Comment by serge-rider** [comment]
What do you mean under "banned"? 
This looks like a regular ssh auth error. Check your user name/password/server address.

If you think that this is a bug in dbeaver - please provide more details.

**Comment by anteldan** [comment]
By "banned" I mean my SSH account gets suspended for a certain time like if there were to many connection attempts with wrong credentials. I cannot even ssh the server in a terminal and my present ssh connections get killed.

With the previous version of DBeaver (6.3.1 something), this hapenned only the first time I tested the tunnel configuration, I had no more possibilities to connect to that machine using SSH for a while which was the time to ask the sysadmin what was wrong with that machine ("nope, everything is fine"), when I got back to my desktop the SSH came back and I could use the SSH tunnel.
I use exactly the same configuration that it was with DBeaver 6.3.1, I just updgraded to 6.3.2 and since then there is no way to connect via SSH tunnel.

I'm using the same private key file in Mysql-workbench and it' OK.
RSA PRIVATE KEY
Proc-Type: 4,ENCRYPTED
DEK-Info: AES-128-CB

**Comment by anteldan** [comment]
To reproduce : 
Start the creation of a new connection:
On the SSH tab check "SSH tunnel"
For "Authentification method" select "public key file"
Select path to private key file (no field for Passphrase is displayed)
Click "Test tunnel configuration"
-> passphrase is not asked and this error is displayed "SSH_MSG_DISCONNECT: 2 Too many authentication failures "
Finish filling the connection's configuration, first tab "General" / Server host: localhost / Port: 3306 / User name: name / etc
Click OK, so the connection is created.

Edit connection (F4 key), look at SSH tab: Passphrase field is displayed, IS empty AND Save password is checked.

I didn't see that Save Password was checked because what matters to me is that Passphrase field is empty. So when I tried to connect I was expecting to fill my passphrase but got rejected.

1. When creating a new connection that uses SSH Tunnel with Public Key Authentification Method, passphrase is not asked when testing tunnel configuration.
2. After this new connection is created, passphrase for Private Key file is stored empty. Checkbox "Save Password" should be unchecked in order to ask user to type passphrase when connecting.

Technically, DBeaver works fine with SSH Tunnel.

**Comment by serge-rider** [comment]
I think it is a duplicate of #7564
You can check the fix in Early Access version (https://dbeaver.io/files/ea)
